client/src/hocs/AuthGuard.js

```diff
@@ -3,42 +3,47 @@ import NavBar from "../components/NavBar";
 
 import { UserContext } from "../context/UserContext";
 
-import { Box, makeStyles } from "@material-ui/core";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+
+import { Box, useMediaQuery } from "@material-ui/core";
 
 const AuthGuard = (ComposedComponent) => {
-  const useStyles = makeStyles((theme) => ({
-    container: {
-      marginTop: "75px",
-    },
-  }));
+  const WithDataComponent = (props) => {
+    const theme = useTheme();
+
+    const isBrowser = useMediaQuery(theme.breakpoints.up("md"));
+
+    const useStyles = makeStyles((theme) => ({
+      container: {
+        // this is the height of the NavBar, adjust or remove this if you change the navbar height or position
+        marginTop: "75px",
+      },
+    }));
 
-  const WithData = (props) => {
     const classes = useStyles();
 
     // data from context
     const { userInfo } = useContext(UserContext);
 
     // check if userInfo is present (user logged in)
     useEffect(() => {
-      if (userInfo) {
-        if (userInfo.isChef) {
-          props.history.push("/chefprofile");
-        } else {
-          props.history.push("/profile");
-        }
-      } else {
-        props.history.push("/");
-      }
+      const redirect = props.history.push;
+
+      const pathName = props.history.location.history;
```

Comment:
A few questions:
What is history.location.history going to refer to?
Is there a consistent path you could redirect them to instead of the above?
Could you use [props.locations](https://reactrouter.com/web/api/location) instead?

Reply:
You should also be able to use the useHistory hook from react-router instead of passing as props